### ISSUE TITLE
[Support] nginx upstream timed out 

### ISSUE THREAD

**Issue by mihob** [beyondcode/laravel-websockets#709]
Maybe someone has an idea.  I have run a load test with artillery and we do not get over approx 1000 connections. Then nginx starts throwing following errors:

`upstream timed out (110: Connection timed out) while proxying upgraded connection`

or 

`upstream timed out (110: Connection timed out) while reading response header from upstream`

Everything should be configured according to https://beyondco.de/docs/laravel-websockets/faq/deploying. Could we have done something wrong here? Unfortunately I can't really assign the error.

This is what the websocket part of our nginx config looks like:

```
location @ws {
  proxy_pass http://127.0.0.1:6001;
  proxy_set_header host $host;
  proxy_read_timeout 300;
  proxy_connect_timeout 120;
  proxy_redirect off;
  # Allow the use of websockets
  proxy_http_version 1.1;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection 'upgrade';
  proxy_set_header Host $host;
  proxy_cache_bypass $http_upgrade;
}
```

**Comment by simonbuehler** [comment]
did you change the event loop or just set `ulimit ` to a higher value with the _correct_ user?

semi-off-topic: did you write a advanced pusher specific artillery script that also joins channels and sends messages? if yes mind sharing or add a PR to the docs?

**Comment by mihob** [comment]
The adjustments for supervisor and ulimit were made. However, the event package was not installed correctly. After that we got over 1000 concurrent users, but at about 3000 connections the following error occurred again:

`upstream timed out (110: Connection timed out) while proxying upgraded connection`

The load of the server was actually always quite low, so I could not really find out where exactly the problem is.

Regarding Artillery I orrientated on this example and played around with the values:
[https://ma.ttias.be/benchmarking-websocket-server-performance-with-artillery/](https://ma.ttias.be/benchmarking-websocket-server-performance-with-artillery/). But it only connects users and subscribes to a channel.
